app/jobs/google/search_keyword_job.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Google
+  class ClientServiceError < StandardError; end
+
+  class SearchKeywordJob < ApplicationJob
+    queue_as :default
+
+    retry_on ClientServiceError, ArgumentError
+
+    def perform(keyword_id)
+      keyword = Keyword.where(id: keyword_id).select(:id, :name, :user_id).take
```

Comment:
I think here we can just make it simple and make less query with `find` as per [sidekiq](https://github.com/mperham/sidekiq/wiki/Best-Practices#2-make-your-job-idempotent-and-transactional) example
```ruby
 keyword = Keyword.find(keyword)
 ```
 
 Then you can update it directly with the object like
 
 ```ruby
keyword.update!(status: status)
 ```
 
 Instead of `Keyword.update keyword_id, status: status` which still will use the `find` method to find the record, then update it.
  What do you think?

Reply:
> That's what I did first, but then observing the logs, it queries the whole object from the DB (including the html column that "can" contain a lot of data).

@malparty How/Where are you logging?

Reply:
yes @junan, I have a tail -f on the dev logs but SQL queries also show up in the console when running the app ;-)

Reply:
Regarding logging, I think it will be the same log and fewer queries if we use `find`. Because currently, it is
doing many tasks for finding the record, in the first part it is query and loads record in an array(actually: `Keyword::ActiveRecord_Relation`), then it filters with `select`, then it picks the first record with `take`
```Ruby
keyword = Keyword.where(id: keyword_id).select(:id, :name, :user_id).take
```
Using `find` we can just find the record(note that at this point there will not `html` or other attribute data as we have not got those data yet)
```ruby
keyword = Keyword.find(keyword_id)
```
Then we can reuse this record where needed. Currently, we are searching this same record with `find` again in the `update_keyword` and `update_keyword_status` with our reusing. 

An even logging also does not matter in this case as Staging/Production, Rails will use [` config.log_level = :info`](https://github.com/malparty/google-search-ruby/blob/develop/config/environments/production.rb#L53), and will not log SQL queries, it just for development envoronment. You can check it by setting `config.log_level = :info` in the `development.rb`

In my idea, we can simplify it like bellow code, which is more simple and efficient
```Ruby
    def perform(keyword_id)
      keyword = Keyword.find(keyword_id)

      return unless keyword

      html_result = ClientService.new(keyword: keyword.name).call

      raise ClientServiceError unless html_result

      update_keyword keyword, ParserService.new(html_response: html_result).call
    rescue StandardError
      update_keyword_status keyword, :failed

      raise
    end

    private

    def update_keyword_status(keyword, status)
      keyword.update!(status: status)
    end

    def update_keyword(keyword, attributes)
      Keyword.transaction do
        # rubocop:disable Rails/SkipsModelValidations
        keyword.result_links.insert_all attributes[:result_links]
        # rubocop:enable Rails/SkipsModelValidations

        keyword.update!(attributes.except(:result_links))
      end
    end
``` 

@malparty What do you think?

Reply:
@junan 
So while I disagree for the "it is doing many tasks for finding the record" (cf picture, only 1 clean select is generated).
You mark a big point by pointing our that Update is actually loading everything again. 

![image](https://user-images.githubusercontent.com/77609814/126303233-59df6bc4-7e79-41c2-87ca-6fa976961453.png)

So I have 2 options:
A. Clean performance option: using `Keyword.where(id: 410).update_all({status: :failed})`
![image](https://user-images.githubusercontent.com/77609814/126304863-88723590-08c3-478d-ba08-f4a4b73a60d2.png)
(and same for updating the attributes)

B. Clean code option, using `find` and voila. :) 

As you pointed the problem out, I will then choose option B. 
In case if was a real project, we would still have observability tools to monitor and be aware if one day it becomes a problem. 

Will be fix soon ;-)

Reply:
Fixed! ;-)